manager/src/main/java/com/redhat/service/bridge/manager/workers/resources/ConnectorWorker.java

```diff
@@ -0,0 +1,242 @@
+    String serviceAccountSecret;
+
+    @Inject
+    ConnectorsDAO connectorsDAO;
+
+    @Inject
+    RhoasService rhoasService;
+
+    @Inject
+    ConnectorsApiClient connectorsApi;
+
+    @Override
+    protected PanacheRepositoryBase<ConnectorEntity, String> getDao() {
+        return connectorsDAO;
+    }
+
+    @Override
+    protected ConnectorEntity runCreateOfDependencies(ConnectorEntity connectorEntity) {
+        // This is idempotent as it gets overridden later depending on actual state
+        connectorEntity = setStatus(connectorEntity, ManagedResourceStatus.PROVISIONING);
+
+        // Step 1 - Create Kafka Topic
+        info(LOGGER,
+                String.format("Creating Kafka Topic for '%s' [%s]",
+                        connectorEntity.getName(),
+                        connectorEntity.getId()));
+        rhoasService.createTopicAndGrantAccessFor(connectorEntity.getTopicName(), RhoasTopicAccessType.PRODUCER);
+
+        // Step 2 - Create Connector
+        info(LOGGER,
+                String.format("Creating Managed Connector for '%s' [%s]",
+                        connectorEntity.getName(),
+                        connectorEntity.getId()));
+        Connector connector = connectorsApi.getConnector(connectorEntity);
+        if (Objects.isNull(connector)) {
+            info(LOGGER,
+                    String.format("Managed Connector for '%s' [%s] not found. Provisioning...",
+                            connectorEntity.getName(),
+                            connectorEntity.getId()));
+            // This is an asynchronous operation so exit and wait for it's READY state to be detected on the next poll.
+            return deployConnector(connectorEntity);
+        }
+
+        // Step 3 - Check it has been provisioned
+        ConnectorStatusStatus status = connector.getStatus();
+        if (Objects.isNull(status)) {
+            return connectorEntity;
+        }
+        if (status.getState() == ConnectorState.READY) {
+            info(LOGGER,
+                    String.format("Managed Connector for '%s' [%s] is ready.",
+                            connectorEntity.getName(),
+                            connectorEntity.getId()));
+
+            // Connector is ready. We can proceed with the deployment of the Processor in the Shard
+            // The Processor will be provisioned by the Shard when it is in ACCEPTED state *and* Connectors are READY (or null).
+            return setReady(connectorEntity);
+        }
+
+        if (status.getState() == ConnectorState.FAILED) {
+            info(LOGGER,
+                    String.format("Managed Connector for '%s' [%s] failed.",
+                            connectorEntity.getName(),
+                            connectorEntity.getId()));
+
+            // Deployment of the Connector has failed. Bubble FAILED state up to ProcessorWorker.
+            return setStatus(connectorEntity, ManagedResourceStatus.FAILED);
+        }
+
```

Comment:
Might be nice to add a bit of logging here to say that we've checked the status of the Connector and it's in status `x`...

Reply:
Done.